pkg/config/config.go

```diff
@@ -71,22 +83,89 @@ func (c *Config) Validate() error {
 	if c.Location == "" {
 		return errors.New("--location is required")
 	}
-	if c.DNSZoneRG != "" || c.DNSZoneSub != "" || c.DNSZoneDomain != "" {
-		if c.DNSZoneRG == "" {
-			return errors.New("--dns-zone-resource-group is required")
-		}
-		if c.DNSZoneSub == "" {
-			return errors.New("--dns-zone-subscription is required")
-		}
-		if c.DNSZoneDomain == "" {
-			return errors.New("--dns-zone-domain is required")
-		}
-	}
 	if c.ConcurrencyWatchdogThres <= 100 {
 		return errors.New("--concurrency-watchdog-threshold must be greater than 100")
 	}
 	if c.ConcurrencyWatchdogVotes < 1 {
 		return errors.New("--concurrency-watchdog-votes must be a positive number")
 	}
+
+	if c.ClusterFqdn == "" {
+		return errors.New("--cluster-fqdn is required")
+	} else {
+		c.ClusterFqdn = hash(c.ClusterFqdn)
+	}
+
+	if dnsZonesString != "" {
+		if err := c.ParseZoneIDs(dnsZonesString); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
+
+func (c *Config) ParseZoneIDs(zonesString string) error {
+
+	c.PrivateZoneConfig = &DnsZoneConfig{}
+	c.PublicZoneConfig = &DnsZoneConfig{}
+
+	DNSZoneIDs := strings.Split(zonesString, ",")
+	for _, zoneId := range DNSZoneIDs {
+		if zoneId == "" {
+			return errors.New("--dns-zone-ids must not contain empty strings")
+		}
+
+		parsedZone, err := azure.ParseResourceID(zoneId)
+		if err != nil {
+			return fmt.Errorf("error while parsing dns zone resource ID %s: %s", zoneId, err)
+		}
+
+		if !strings.EqualFold(parsedZone.Provider, "Microsoft.Network") {
+			return fmt.Errorf("invalid resource provider %s from zone %s: resource ID must be a public or private DNS Zone resource ID from provider Microsoft.Network", parsedZone.Provider, zoneId)
+		}
+
+		if strings.EqualFold(parsedZone.ResourceType, PrivateZoneType) {
+			// it's a private zone
+			if err := validateZoneId(parsedZone, c.PrivateZoneConfig.Subscription, c.PrivateZoneConfig.ResourceGroup); err != nil {
+				return err
+			}
+
+			c.PrivateZoneConfig.Subscription = parsedZone.SubscriptionID
+			c.PrivateZoneConfig.ResourceGroup = parsedZone.ResourceGroup
+			c.PrivateZoneConfig.ZoneIds = append(c.PrivateZoneConfig.ZoneIds, zoneId)
+
+		} else if strings.EqualFold(parsedZone.ResourceType, PublicZoneType) {
```

Comment:
Switch statement is more effective go than if elif else here

Reply:
switched! (pun fully intended)